lib/std/os.zig

```diff
@@ -660,8 +660,35 @@ pub fn kill(pid: pid_t, sig: u8) KillError!void {
     }
 }
 
-/// Exits the program cleanly with the specified status code.
-pub fn exit(status: u8) noreturn {
+/// The type of the value that is passed to the operating system when exiting.
+/// See: `exit`.
+pub const ExitStatus = switch (builtin.os.tag) {
+    .plan9 => [:0]const u8,
+    .uefi => std.os.uefi.Status,
+    // TODO: `.wasi => u7,`
+    //       This involves updating all `main`s that use `u8` to use std.os.ExitStatus.
+    //       Also, wasmtime says the actual range is "[0..126)". Use `@Int(0, 125)` when possible.
+    else => u8,
+};
+
+/// A general exit status indicating success.
+/// This can be used as a default or when details beyond the fact that execution succeeded are unavailable.
+pub const exit_status_success: ExitStatus = switch (builtin.os.tag) {
+    .plan9 => "",
+    .uefi => std.os.uefi.Status.Success,
+    else => 0,
+};
+
+/// A general exit status indicating failure.
+/// This can be used as a default or when details beyond the fact that execution failed are unavailable.
+pub const exit_status_failure: ExitStatus = switch (builtin.os.tag) {
+    .plan9 => "failure",
```

Comment:
CC @g-w1 is there a convention? Something better we can use here as a general exit status indicating failure without any specifics?

Reply:
I think a convention is exits("error") as a bunch of the commands here (http://git.9front.org/plan9front/plan9front/ea1d75e67581f9a65abddbe2f03af70ea8eb1e1a/sys/src/cmd/f.html) use it. But a lot also use specifics. So I think "error" might be better. But failure works fine too. I think generally, it would be better to `exit(@errorName(err))` if main is a !void function instead of manually doing either failure or success.

Reply:
Ah, I see! A lot of places just say "error" instead of "failure". I'll use that then!
Yes, of course specifics are always going to be better than "error". What you're suggesting is a great idea! I'll do that in `start.zig`.

Reply:
Though, we already do `std.log.err("{s}", .{@errorName(err)});` when there's an error union returned from `main`. Should we maybe not do the `std.log.err` on Plan 9 and exit with `"{s}", .{@errorName(err)}`? Or should we do both? Either way I agree with `exits(@errorName(err))`.

Reply:
I'm not sure. I think `std.log.err("{s}", .{@errorName(err)});` would still be nice in addition to exiting with the error. If it turns out that it is redundant, we can always remove it.